### ISSUE TITLE
Messages not translated when running in production environment with webpack

### ISSUE THREAD

**Issue by k-vekos** [robisim74/angular-l10n#223]
**I'm submitting a...**
```
[x] bug report
[ ] feature request
[ ] support request
```
Using Angular v5 with angular-l10n v4.2.0, angular-l10n will leave strings unlocalized when in production build only.

**Steps to reproduce the behavior**

Here is my `webpack.config.js`:
```
const path = require('path');
const webpack = require('webpack');
const AngularCompilerPlugin = require('@ngtools/webpack').AngularCompilerPlugin;
const CheckerPlugin = require('awesome-typescript-loader').CheckerPlugin;
const TSLintPlugin = require('tslint-webpack-plugin');
const ExtractTextPlugin = require('extract-text-webpack-plugin');

module.exports = (env) => {
    const isDevBuild = !(env && env.prod);
    const bundleOutputDir = './wwwroot/dist';
    const bundleConfig = {
        stats: { modules: false },
        context: __dirname,
        resolve: { extensions: ['.js', '.ts'] },
        entry: { 'main': './ClientApp/boot.ts' },
        output: {
            filename: '[name].js',
            publicPath: 'dist/', // Webpack dev middleware, if enabled, handles requests for this URL prefix
            path: path.join(__dirname, bundleOutputDir)
        },
        module: {
            rules: [
                { test: /\.ts$/, include: /ClientApp/, use: isDevBuild ? ['awesome-typescript-loader?silent=true', 'angular2-template-loader'] : '@ngtools/webpack' },
                { test: /\.html$/, use: 'html-loader?minimize=false' },
                { test: /\.(woff|woff2|ttf|eot|ico)$/, use: { loader: 'file-loader', options: { useRelativePath: true, publicPath: '' } } },
                { test: /\.css$/, include: path.join(__dirname, './ClientApp/app/assets/css'), use: ExtractTextPlugin.extract({ use: isDevBuild ? 'css-loader' : 'css-loader?minimize' }) },
                { test: /\.css$/, exclude: path.join(__dirname, './ClientApp/app/assets/css'), use: ['to-string-loader', isDevBuild ? 'css-loader' : 'css-loader?minimize'] },
                { test: /\.(png|jpg|jpeg|gif|svg)$/, use: 'url-loader?limit=25000' }
            ]
        },
        plugins: [
            new CheckerPlugin(),
            new webpack.DllReferencePlugin({
                context: __dirname,
                manifest: require('./wwwroot/dist/vendor-manifest.json')
            }),
            new ExtractTextPlugin("[name].css")  
        ].concat(isDevBuild ? [
            // Plugins that apply in development builds only
            new webpack.SourceMapDevToolPlugin({
                filename: '[file].map', // Remove this line if you prefer inline source maps
                moduleFilenameTemplate: path.relative(bundleOutputDir, '[resourcePath]') // Point sourcemap entries to the original file locations on disk
            })
        ] : [
                // Plugins that apply in production builds only
                new webpack.optimize.UglifyJsPlugin({
                    // Drop console statements
                    compress: { drop_console: true }
                }),
                new AngularCompilerPlugin({ 
                    tsConfigPath: './tsconfig.json',
                    entryModule: path.join(__dirname, 'ClientApp/app/app.module#AppModule')
                })
            ])
    };

    return [bundleConfig];
};
```

So basically, if `isDevBuild` is true it is working fine, otherwise `'StringName' | translate` returns `'StringName'` even when running browser in original English language.

I also noticed that in production env `LocaleService.getCurrentLocale()` is returning `undefined` instead of the expected `'en-US'`.

**Comment by robisim74** [comment]
@k-vekos Are you implementing `ngOnInit` method in your components that use translate pipe even if it is empty? It is required for AoT build: https://robisim74.github.io/angular-l10n/spec/getting-the-translation/#messages

**Comment by k-vekos** [comment]
@robisim74 Thank you! I found the problem. I thought I had AoT disabled but I didn't. I changed my webpack.config.js to add `skipCodeGeneration: true` though and now it's working:

```
new AngularCompilerPlugin({ 
                    tsConfigPath: './tsconfig.json',
                    entryModule: path.join(__dirname, 'ClientApp/app/app.module#AppModule'),
                    skipCodeGeneration: true
                })
```

If I turn it back on though, is all I have to do is implement `ngOnInit` in all my component classes? Can the implementations just be empty? Is it only for those components which use the pipes?

Thanks!
